dev/com.ibm.ws.jaxrs.2.0_fat/test-applications/simpleJson/src/com/ibm/ws/jaxrs/fat/simpleJson/JaxrsJsonClientTestServlet.java

```diff
@@ -0,0 +1,114 @@
+import javax.ws.rs.client.ClientBuilder;
+import javax.ws.rs.client.Entity;
+import javax.ws.rs.client.ResponseProcessingException;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+
+import org.apache.http.HttpResponse;
+import org.apache.http.client.HttpClient;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.entity.StringEntity;
+import org.apache.http.impl.client.DefaultHttpClient;
+import org.junit.After;
+import org.junit.Test;
+
+import componenttest.app.FATServlet;
+
+@WebServlet(urlPatterns = "/JaxrsJsonClientTestServlet")
+public class JaxrsJsonClientTestServlet extends FATServlet {
+
+    private static final long serialVersionUID = 4563445389586844836L;
+
+    final static String URI_CONTEXT_ROOT = "http://localhost:" + Integer.getInteger("bvt.prop.HTTP_default") + "/simpleJson/";
+
+    private static Client client;
+
+    // needed to make requests with invalid json to the server
+    private static HttpClient httpClient;
+
+    @Override
+    public void init() throws ServletException {
+        client = ClientBuilder.newClient();
+//      httpClient = new HttpClientBuilder.create().build();
+        httpClient = new DefaultHttpClient();
+    }
+
+    @After
+    private void teardown() {
+        client.close();
+        httpClient.getConnectionManager().shutdown();
+    }
+
+    @Test
+    public void simpleTest() throws Exception {
+        Car corvette = new Car();
+        corvette.color = "red";
+        corvette.make = "Chevrolet";
+        corvette.model = "Corvette";
+        corvette.year = 2014;
+
+        Response response = client.target(URI_CONTEXT_ROOT)
+                        .path("simpleresource/post")
+                        .request(MediaType.APPLICATION_JSON_TYPE)
+                        .post(Entity.json(corvette));
+        assertEquals(200, response.getStatus());
+        String actual = response.readEntity(String.class);
+
+        assertEquals("2014 Chevrolet Corvette", actual);
+    }
+
+    @Test
+    public void sendInvalidJson() throws Exception {
+
+        HttpPost post = new HttpPost(URI_CONTEXT_ROOT + "simpleresource/post");
+        StringEntity entity = new StringEntity("invalid");
+        post.setEntity(entity);
+        post.addHeader("Content-Type", "application/json");
+
+        HttpResponse response = httpClient.execute(post);
+        assertEquals(400, response.getStatusLine().getStatusCode());
+    }
+
+    @Test
+    public void recieveInvalidJson() throws Exception {
+
+        Response response = client.target(URI_CONTEXT_ROOT)
+                        .path("badsimpleresource/badresponse")
+                        .request(MediaType.APPLICATION_JSON_TYPE)
+                        .get();
+
+        boolean exception = false;
+        try {
+            response.readEntity(Foo.class);
+        } catch (ResponseProcessingException e) {
+            exception = true;
+        }
+
+        if(!exception) {
+            fail();
+        }
```

Comment:
You could remove this and just call `fail()` in the try block after `response.readEntity(Foo.class);`